frontend/src/Editor/Editor.jsx

```diff
@@ -30,11 +29,16 @@ import { WidgetManager } from './WidgetManager';
 import Fuse from 'fuse.js';
 import config from 'config';
 import queryString from 'query-string';
+import toast from 'react-hot-toast';
+import { cloneDeep, isEqual, isEmpty } from 'lodash';
+import produce, { enablePatches, setAutoFreeze, applyPatches } from 'immer';
 import Logo from './Icons/logo.svg';
 import EditIcon from './Icons/edit.svg';
 import MobileSelectedIcon from './Icons/mobile-selected.svg';
 import DesktopSelectedIcon from './Icons/desktop-selected.svg';
 
+setAutoFreeze(false);
```

Comment:
What happens if we set this, it should be a safeguard to not allow any modifications?  🤔

Reply:
Actually what immer does is it uses Object.freeze and gives out an immutable object, we are using immer for the `patches` functionality currently, 

this disables the immutability feature as introducing that at current stage is a slightly bigger change

Reply:
Oh I see, yeah by reading immer docs I got that impression. Thought we can leverage immutability here as well. Sure can do later, just curious.